src/components/Table/tableReducer.js

```diff
@@ -22,6 +22,8 @@ import {
   TABLE_COLUMN_ORDER,
   TABLE_REGISTER,
   TABLE_SEARCH_APPLY,
+  TABLE_DATA_SAVE,
+  TABLE_DATA_UNDO,
```

Comment:
I'm on the fence about building this into the table and exporting these stateful reducers. Would these be used? It feels like this would be implemented at the product level to hook up to a data source or central state (redux, mobx, apollo, etc) via their own reducers or hooks. 

In it's place, we could provide any necessary callbacks or props to enable this behavior (including the "undo" state).

@bryancboyd @davidicus Do you have some thoughts here?

Reply:
Does this affect my ability to hook into the save and undo events coming from the table?  ie, I need to be able to register listeners that are called when save and undo are clicked.  The whole messaging system of the table is extremely confusing to me, so, I have no idea if this is needed to achience the simple callback or not.  But, we do need to get notified when these things happen in the table as a result of the user clicking save or cancel or undo.  I'm not using the stateful table, so not sure if this is ONLY for the stateful implementation.

Reply:
@tay1orjones I agree with your comment. Would like to hear what @bryancboyd thinks. @stuckless These reducers are for the stateful implementation only I believe.